lib/token.js

```diff
@@ -0,0 +1,285 @@
+'use strict';
+
+const types = require('./types');
+const util = require('util');
+
+const _Murmur3TokenType = types.dataTypes.getByName('bigint');
+const _RandomTokenType = types.dataTypes.getByName('varint');
+const _OrderedTokenType = types.dataTypes.getByName('blob');
+
+/**
+ * Represents a token on the Cassandra ring.
+ *
+ * @property value The raw value of the token.  Type depends on
+ *                 partitioner implementation. 
+ * 
+ */
+class Token {
+  constructor(value, tokenizer) {
+    this.value = value;
+    Object.defineProperty(this, '_tokenizer', { value: tokenizer, enumerable: false});
```

Comment:
I think that, instead of `Token#compare()` calling `Tokenizer#compare()` each `Token` implementation should have a `compare()` implementation.
We can remove `Tokenizer#compare()` as it's not being used anymore by `buildTokens()`.

Besides that, I think we should avoid mixing `Token#toString()` with `Tokenizer#stringify()`: `Token#toString()` is a string representation of a token (same as C* tools) and `Tokenizer#stringify()` is a workaround only present on the nodejs driver suited for allowing string hashes of tokens (object properties can only by strings). I propose we implement `Token#toString()` and the base tokenizer uses it default implementation for `stringify()` but each `Tokenizer` implementation can override that behavior to provide the fastest hash for that token (most notably on murmur3).

If we implement those changes, we can get rid of the `Tokenizer` dependency in the `Token` constructor.

Reply:
Both great ideas, will do!